### PR TITLE
Implementation of less_equal function, conforming to Array API standard

### PR DESCRIPTION
**Pull request by georg-wolflein**
This is the implementation of issue #272.

Although all tests are passing on my local machine (numpy, jax, torch, tensorflow), I'd like to draw attention to two things:
1. I needed to modify `builtin_le` in `ivy/functional/ivy/core/array_builtins.py` to return the result of `ivy.less_equal` instead of `x.__le__` to get the tensorflow tests to pass. 
2. The tensorflow and pytorch implementation require some dtype casting which is similar for the `equal` function, so I refactored it into the `_compare` function, as I presume similar logic will be needed in the future for functions like `greater_equal`(#198, #220, #236, #126, etc.).

### CODE REVIEW

**Comment by djl11** [comment]
Thanks a lot for your contribution. Could you update the methods to follow the formatting explained in the [contributor guide](https://lets-unify.ai/ivy/contributing.html) in the [docs](https://lets-unify.ai/ivy/)? I also go through some examples in the Array API [tutorial series](https://www.youtube.com/watch?v=HNV4eI0zp4M&list=PLwNuX3xB_tv96FaEUmTirlQLIsnUgvtTl) on YouTube.

Essentially:

- All methods should be defined with `def func_name` explicitly, no lambdas or direct binding.
- All Ivy methods should have type-hints and doc strings
- All backend methods should have type-hints, but no doc strings.
- All Array API methods should be moved to the correct sub-modules and files.

Please check the contributor guide in the docs for more details. Thanks!

Additionally, it may be necessary to resolve new conflicts with master since the PR was made. As a guide for resolving conflicts with upstream, I have created this short [YouTube series](https://www.youtube.com/watch?v=KCMceLBBlgs&list=PLwNuX3xB_tv8Gf2sELkXSqCCWcUq-AbrG).
Once there are no conflicts with upstream and the formatting is all correct, I'll then take a look at the tests and see if it's ready to be merged :)

**Review by djl11 — CHANGES_REQUESTED** [review]
Please make changes as explained in the earlier comment :)

**Review by djl11 — CHANGES_REQUESTED** [review]
Please reformat the code as describe in the [contributor guide](https://lets-unify.ai/ivy/contributing.html) in the [docs](https://lets-unify.ai/ivy/), and also as explained in this [YouTube series](https://www.youtube.com/watch?v=HNV4eI0zp4M&list=PLwNuX3xB_tv96FaEUmTirlQLIsnUgvtTl).

**Comment by georg-wolflein** [comment]
Hi @djl11, thanks for your comments. Will implement the changes later today.

File: ivy/array/__init__.py [hunk 1]
```diff
@@ -275,11 +275,7 @@ def __lt__(self, other):
 
     @_native_wrapper
     def __le__(self, other):
-        other = to_native(other)
-        res = ivy.builtin_le(self._data, other)
-        if res is NotImplemented:
-            return res
-        return to_ivy(res)
+        return ivy.less_equal(self._data, other)
```
Comment (georg-wolflein) [inline ivy/array/__init__.py:278]: I couldn't use `ivy.builtin_le` because that fails one of the tensorflow Array API tests

**Comment by 1Doomdie1** [comment]
Hey georg-wolflein, thank you for contributing!
Next time after you made the changes requested by a ivy team member please request a "Code Review".
I've requested one for you now but please keep this in mind!
If you fail to do so the PR will remain unchecked and will be closed automatically!

**Comment by djl11** [comment]
Thanks!

**Pull request merged**